app/src/main/java/de/danoeh/antennapod/activity/OnlineFeedViewActivity.java

```diff
@@ -509,11 +510,22 @@ private void handleUpdatedFeedStatus(Feed feed) {
                 viewBinding.subscribeButton.setEnabled(true);
                 viewBinding.subscribeButton.setText(R.string.open_podcast);
                 if (didPressSubscribe) {
+                    didPressSubscribe = false;
+                    if (UserPreferences.isEnableAutodownload()) {
+                        Feed feed1 = DBReader.getFeed(getFeedId(feed));
+                        FeedPreferences feedPreferences = feed1.getPreferences();
+                        feedPreferences.setAutoDownload(viewBinding.autoDownloadCheckBox.isChecked());
+                        feed1.savePreferences();
+                    }
                     openFeed();
                 }
             } else {
                 viewBinding.subscribeButton.setEnabled(true);
                 viewBinding.subscribeButton.setText(R.string.subscribe_label);
+                if (UserPreferences.isEnableAutodownload()) {
+                    viewBinding.autoDownloadCheckBox.setChecked(true);
```

Comment:
I think it would be better to do this in XML, too. On my device, the line is executed again just before the settings are updated. That results in all feeds having autodl enabled, regardless of what you do with the checkbox